xwiki-platform-core/xwiki-platform-repository/xwiki-platform-repository-server-api/src/main/java/org/xwiki/repository/script/RepositoryScriptService.java

```diff
@@ -68,14 +68,15 @@ private void setError(Exception e)
 
     /**
      * Get the error generated while performing the previously called action.
-     * 
+     *
      * @return an eventual exception or {@code null} if no exception was thrown
      */
     public Exception getLastError()
     {
         return (Exception) this.execution.getContext().getProperty(REPOSITORYERROR_KEY);
     }
 
+    @Deprecated(since = "16.6.0RC1")
```

Comment:
Thanks! Btw if the method was really unused and not useful, it might not be worth the struggle of moving it to legacy. Would you be ok to just remove if (if you say yes, I'll open a forum proposal)?

Reply:
I don't really agree with making this API deprecated, it's in no way a fix for what the issue describes. The only thing that can be seen as a fix in this pull request is the fact that you removed the UI which calls it.

If there is anything to change on Java side is fixing the bug in RepositoryManager#validateExtensions (which is mostly dues to a behavior change in Hibernate over the years which "optimized" a bit by return type depending on the number of element in the select).

Reply:
ok, so in summary:
- create a separate improvement issue for the removal of the UI part (but keep the `validateExtensions` method even if unused).
- fix the error initially reported error fromhttps://jira.xwiki.org/browse/XWIKI-17548 in this PR.

Does it match what you had in mind?

Reply:
Honestly I would have removed the UI and fixed the bug in the same pull request. But having an issue dedicated to the removing of the admin section would probably make sense.

Reply:
Depends, do we consider the removal a bug fix or an improvement? Based on this choice, it will not apply on the same branches.

Reply:
Hard to make the removing of an admin section pass as a bugfix in general, it kind of being perceived as such in your PR because it makes a bug a bit more hidden, but it's cheating a bit.

Now I feel that fixing the bug would have been faster than this whole discussion :)

Reply:
Yep, sorry about that. I hope the new change is ok now.